### ISSUE TITLE
could install from source on ubuntu 18.04 'Solaar: missing required package 'python-pyudev'

### ISSUE THREAD

**Issue by yifita** [pwr-Solaar/Solaar#438]
Hi,

I want to install from source to support **anywhere master 2s**. 

I was using conda python. In order to allow system wide installation, I removed the conda path, so now 
```
$ which python
$ /usr/bin/python
$ python --version
$ Python 3.6.5
```

I made sure to install python3-pyudev 
```
$ sudo apt install python3-pyudev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
python3-pyudev is already the newest version (0.21.0-1).
0 upgraded, 0 newly installed, 0 to remove and 22 not upgraded.
```

Build and install was successful with python3

```
$ sudo ./setup.py install --dry-run
running install
running build
running build_py
creating build/lib
creating build/lib/hidapi
copying lib/hidapi/udev.py -> build/lib/hidapi
copying lib/hidapi/__init__.py -> build/lib/hidapi
copying lib/hidapi/hidconsole.py -> build/lib/hidapi
creating build/lib/logitech_receiver
copying lib/logitech_receiver/common.py -> build/lib/logitech_receiver
copying lib/logitech_receiver/receiver.py -> build/lib/logitech_receiver
copying lib/logitech_receiver/settings.py -> build/lib/logitech_receiver
copying lib/logitech_receiver/hidpp10.py -> build/lib/logitech_receiver
copying lib/logitech_receiver/descriptors.py -> build/lib/logitech_receiver
copying lib/logitech_receiver/i18n.py -> build/lib/logitech_receiver
copying lib/logitech_receiver/__init__.py -> build/lib/logitech_receiver
copying lib/logitech_receiver/hidpp20.py -> build/lib/logitech_receiver
copying lib/logitech_receiver/settings_templates.py -> build/lib/logitech_receiver
copying lib/logitech_receiver/notifications.py -> build/lib/logitech_receiver
copying lib/logitech_receiver/base_usb.py -> build/lib/logitech_receiver
copying lib/logitech_receiver/listener.py -> build/lib/logitech_receiver
copying lib/logitech_receiver/base.py -> build/lib/logitech_receiver
copying lib/logitech_receiver/status.py -> build/lib/logitech_receiver
copying lib/logitech_receiver/special_keys.py -> build/lib/logitech_receiver
creating build/lib/solaar
copying lib/solaar/upower.py -> build/lib/solaar
copying lib/solaar/i18n.py -> build/lib/solaar
copying lib/solaar/__init__.py -> build/lib/solaar
copying lib/solaar/async.py -> build/lib/solaar
copying lib/solaar/listener.py -> build/lib/solaar
copying lib/solaar/gtk.py -> build/lib/solaar
copying lib/solaar/configuration.py -> build/lib/solaar
creating build/lib/solaar/ui
copying lib/solaar/ui/pair_window.py -> build/lib/solaar/ui
copying lib/solaar/ui/window.py -> build/lib/solaar/ui
copying lib/solaar/ui/config_panel.py -> build/lib/solaar/ui
copying lib/solaar/ui/tray.py -> build/lib/solaar/ui
copying lib/solaar/ui/__init__.py -> build/lib/solaar/ui
copying lib/solaar/ui/notify.py -> build/lib/solaar/ui
copying lib/solaar/ui/icons.py -> build/lib/solaar/ui
copying lib/solaar/ui/about.py -> build/lib/solaar/ui
copying lib/solaar/ui/action.py -> build/lib/solaar/ui
creating build/lib/solaar/cli
copying lib/solaar/cli/config.py -> build/lib/solaar/cli
copying lib/solaar/cli/unpair.py -> build/lib/solaar/cli
copying lib/solaar/cli/pair.py -> build/lib/solaar/cli
copying lib/solaar/cli/__init__.py -> build/lib/solaar/cli
copying lib/solaar/cli/show.py -> build/lib/solaar/cli
running install_lib
running install_data
running install_egg_info
Removing /usr/local/lib/python3.6/dist-packages/solaar-0.9.2.egg-info
Writing /usr/local/lib/python3.6/dist-packages/solaar-0.9.2.egg-info
```

But when I run solaar
```
$ solaar
Solaar: missing required package 'python-pyudev'
```

BTW I did actually follow the instruction to install `python-pyudev`, since 18.04 stopped using python2, this actually installed python2. But the problem resists!

**Comment by AcerLaurinum** [comment]
I had this same error and for whatever reason doing `sudo python solaar` instead fixed it.

**Comment by yifita** [comment]
`which python` is python2 or python3?

**Comment by paulosoares** [comment]
try:
$ sudo apt install python3-pyudev
$ sudo pip install pyudev

**Comment by dmichelin** [comment]
sudo apt install python3-pyudev worked for me

**Comment by pfps** [comment]
Is this still a problem?  If so, maybe the documentation needs to be updated.

**Comment by emwalker** [comment]
This is still a problem, after upgrading to Ubuntu 20.04.

**Comment by lubacien** [comment]
this is still a problem

**Comment by pfps** [comment]
What did you do that caused the problem?  Which distribution and version are you using?

**Comment by pfps** [comment]
It might be that the python that you run is a different version from the python that the system (and root) uses.

**Comment by pcdro** [comment]
I had the same problem today after installing the tool for the first time based on this question of Stack Overflow: https://askubuntu.com/a/1309454/736957

After running @dmichelin suggestion of installing `python3-pyudev` the following flow worked:

```bash
sudo apt-get update
sudo apt install solaar
sudo apt install python3-pyudev
solaar
```

**Comment by meniluca** [comment]
solved with installing 1.0.4 and running:

```pip install pycairo PyGObject```
